paper_framework_sources/p1897_initial_executor_algorithms_for_cpp23.md

```diff
@@ -25,56 +35,71 @@ toc: false
 It is open to discussion how we achieve this and that is an independent topic, beyond the scope of this paper.
 
 ## Summary
-Starting with [@P0443R11] as a baseline we have the following customization points:
+Starting with [@P0443R13] as a baseline we have the following customization points:
 
- * `execute(executor, invocable) -> void`
+ * `connect(sender, receiver) -> operation_state`
+ * `start(operation_state) -> void`
  * `submit(sender, receiver) -> void`
  * `schedule(scheduler) -> sender`
+ * `execute(executor, invocable) -> void`
  * `set_done`
  * `set_error`
  * `set_value`
 
 and the following Concepts:
 
- * `executor`
  * `scheduler`
- * `callback_signal`
- * `callback`
+ * `receiver`
+ * `receiver_of`
  * `sender`
+ * `sender_to`
+ * `typed_sender`
+ * `operation_state`
+ * `executor`
+ * `executor_of`
 
 We propose immediately discussing the addition of the following algorithms:
 
  * `just(v)`
    * returns a `sender` of the value `v`
- * `just_via(sch, v)`
-   * a variant of the above that embeds the `via` algorithm
- * `via(s, sch)`
+ * `just_on(sch, v)`
+   * a variant of the above that embeds the `on` algorithm
+ * `on(s, sch)`
    * returns a sender that propagates the value or error from `s` on `sch`'s execution context
  * `sync_wait(s)`
    * blocks and returns the value type of the sender, throwing on error
  * `when_all(s...)`
    * returns a sender that completes when all Senders `s...` complete, propagating all values
- * `indexed_for(s, policy, rng, f)`
-   * returns a sender that applies `f` for each element of `rng` passing that element and the values from the incoming sender, completes when all `f`s complete propagating s's values onwards
+ <!--* `indexed_for(s, policy, rng, f)`
+   * returns a sender that applies `f` for each element of `rng` passing that element and the values from the incoming sender, completes when all `f`s complete propagating s's values onwards-->
  * `transform(s, f)`
    * returns a sender that applies `f` to the value passed by `s`, or propagates errors or cancellation
- * `bulk_transform(s, f)`
-   * returns a sender that applies `f` to each element in a range sent by `s`, or propagates errors or cancellation
+ <!--* `bulk_transform(s, f)`
+   * returns a sender that applies `f` to each element in a range sent by `s`, or propagates errors or cancellation-->
  * `handle_error(s, f)`
-   * returns a sender that applies `f` to an error passed by `s`, ignoring the values or cancellation
+   * Creates an async scope where an error propagated by `s` is available for the duration of another async operation `f`.
+     Value and cancellation propagate unmodified.
+ * `share(s)`
+   * Eagerly submits `s` and returns a sender that may be used more than once, propagating the value by reference.
+ * `let(s, f)`
+   * Creates an async scope where the value propagated by `s` is available for the duration of another async operation `f`.
+     Error and cancellation signals propagate unmodified.
```

Comment:
some others to consider:

repeat_effect() implemented in libunifex
replay_first_after_last_effect() implemented in libunifex as sequence()
take_until() implemented in libunifex as stop_when()
transform_done() implemented in libunifex (completes the let/handle_error triumvirate)
retry() - does not require void producing source sender
sequence_effects() this would require that all senders were void producing (this directly models the common uses of libunifex's sequence())
sequenced_all() is the same as when_all() but runs the senders in sequence. might motivate rename of when_all() to unsequenced_all()

time - requires TimeScheduler, schedule_at(), and now():

schedule_after() can be implemented in terms of schedule_at() and now()
interval()

composed:

timeout() can be composed from schedule_at(), now(), take_until(),  and transform_done()
filter() can be composed from transform_value() aka let()